scipy/spatial/distance.py

```diff
@@ -1994,8 +1996,8 @@ def cdist(XA, XB, metric='euclidean', p=2, V=None, VI=None, w=None):
     XB = np.asarray(XB, order='c')
 
     # The C code doesn't do striding.
-    XA = _copy_array_if_base_present(_convert_to_double(XA))
-    XB = _copy_array_if_base_present(_convert_to_double(XB))
```

Comment:
It also seems inconsistent to completely remove the conversion from ``cdist``, but leave it in for ``pdist``.